### PR TITLE
fix: `ComponentContext.label` not updated to reflect component models

### PR DESCRIPTION
**Pull request by GeomKid**
## About

This pull request is about fixing the bug of label, which causes the following error:
```py
Traceback (most recent call last):
  File "C:\Users\Asus\Documents\GitHub\Python-Slash-cmd\newcogs\cstop.py", line 795, in choice0
    ctx.label
  File "C:\Users\Asus\Documents\GitHub\Python-Slash-cmd\.venv\lib\site-packages\interactions\client\context.py", line 680, in label
    for component in action_row["components"]:
TypeError: 'ActionRow' object is not subscriptable
```
Original Code:
```py
@extension_component("drjoin")
async def choice0(self, ctx: ComponentContext):
    ctx.label 
```
## Checklist

- [x] The ``pre-commit`` code linter has been run over all edited files to ensure the code is linted.
- [x] I've ensured the change(s) work on `3.8.6` and higher.

I've made this pull request: (check all that apply)
  - [x] For the documentation
  - [ ] To add a new feature
  - [ ] As a general enhancement
  - [ ] As a refactor of the library/the library's code
  - [x] To fix an existing bug
  - [ ] To resolve #ISSUENUMBER

This is:
  - [ ] A breaking change

### CODE REVIEW

File: interactions/client/context.py [hunk 1]
```diff
@@ -738,7 +738,19 @@ def custom_id(self) -> Optional[str]:
 
     @property
     def label(self) -> Optional[str]:
+        """
+        Gives the label of button
```
Comment (Toricane) [inline interactions/client/context.py]:
```suggestion
        The label of the interacted component.
```

File: interactions/client/context.py [hunk 2]
```diff
@@ -738,7 +738,19 @@ def custom_id(self) -> Optional[str]:
...
+        :rtype: Optional[str]
+        """
+        if not self.data.component_type == ComponentType.BUTTON:
+            return None
```
Comment (Toricane) [inline interactions/client/context.py]: Don't selects have labels too? Remove this

File: interactions/client/context.py [hunk 3]
```diff
@@ -738,7 +738,19 @@ def custom_id(self) -> Optional[str]:
...
+        if self.message is None:
+            return None
```
Comment (Toricane) [inline interactions/client/context.py]:
```suggestion
            return
```

File: interactions/client/context.py [hunk 4]
```diff
@@ -738,7 +738,19 @@ def custom_id(self) -> Optional[str]:
...
+        if self.message.components is None:
+            return None
```
Comment (Toricane) [inline interactions/client/context.py]:
```suggestion
            return
```

File: interactions/client/context.py [hunk 5]
```diff
@@ -738,7 +738,19 @@ def custom_id(self) -> Optional[str]:
...
         for action_row in self.message.components:
-            for component in action_row["components"]:
-                if component["custom_id"] == self.custom_id and component["type"] == 2:
-                    return component.get("label")
+            for component in action_row.components:
+                if component.custom_id == self.custom_id:
+                    return component.label
+        else:
+            return None
```
Comment (Toricane) [inline interactions/client/context.py]: Don't need this as the function will return None when the loop ends

**Review by Toricane — CHANGES_REQUESTED** [review]
A few changes

File: interactions/client/context.py (hunk 2 above)
Comment (GeomKid) [reply interactions/client/context.py]: I checked selects don't have labels

Comment (Toricane) [reply interactions/client/context.py]: Oops i confused it with the custom_id

File: interactions/client/context.py [hunk 6]
```diff
@@ -738,7 +738,17 @@ def custom_id(self) -> Optional[str]:
 
     @property
     def label(self) -> Optional[str]:
+        """
+        The label of the interacted component.
```
Comment (Toricane) [inline interactions/client/context.py]: One last suggestion
```suggestion
        The label of the interacted button.
```

**Review by Toricane — CHANGES_REQUESTED** [review]
One last suggestion

**Review by Toricane — APPROVED** [review]
LGTM!

**Pull request merged**